src/client/cockpit-client-ssh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+
+import base64
+import json
+import logging
+import os
+import shutil
+import sys
+
+
+def send_control(**msg):
+    logging.debug(msg)
+    fp = sys.stdout.buffer
+    msg = f'\n{json.dumps(msg)}\n'.encode('utf-8')
+    # yes, we're writing to stdin: it's a socket, and for askpass, ssh captures stdout
+    os.write(0, f'{len(msg)}\n'.encode('utf-8') + msg)
+
+
+def recv_control():
+    # using buffered IO is safe here: we only get one message sent at a
+    # time, and then it's our turn to say something before the next one
+    # will be sent.  therefore, it's impossible to over-read.
+    fp = sys.stdin.buffer
+    length = int(fp.readline())
+    # take advantage of empty channel name being json whitespace
+    return json.loads(fp.read(length))
+
+
+def do_authorize(challenge='*', **kwargs):
+    send_control(command='authorize', challenge=challenge, cookie='-', **kwargs)
```

Comment:
Will this ever be used for external ws? If so, then this should really generate a random cookie and validate it in the response.

Reply:
No.  It won't.  But I'll generate the random cookie anyway.